File: Java/NetBeans/DSNV/src/dsnv/NVHD.java
package dsnv;

public class NVHD extends NV{
    protected double luong;
    protected String loaiHD;
    
    public NVHD(String hoTen, int phong, double luong, String loaiHD) {
        super(hoTen, phong);
        this.luong = luong;
        this.loaiHD = loaiHD;
    }
    
    public void hienThi() {
        super.hienThi();
        System.out.println("Luong(trieu VND): " + luong);
        System.out.println("Loai hop dong: " + loaiHD);
    }
    
    public String loaiNV() {
        return "HD";
    }
    
    public String layLoaiHD() {
        return loaiHD;
    }
    
    public double layLuong() {
        return luong;
    }
    
    public boolean laNVHDDH() {
        return loaiHD.equalsIgnoreCase("DH");
    }
}

File: C#/DSNV/DSNV/NVHD.cs
namespace DSNV
{
    public class NVHD : NV
    {
        private float luong;
        private bool loaiHD;

        protected bool LoaiHD
        {
            get { return loaiHD; }
            set { loaiHD = value; }
        }

        protected float Luong
        {
            get { return luong; }
            set { luong = value; }
        }

        public NVHD(string hoTen, int phong, float luong, bool loaiHD) : base(hoTen, phong)
        {
            this.luong = luong;
            this.loaiHD = loaiHD;
        }

        public override void hienThi()
        {
            base.hienThi();
            Console.WriteLine($"Luong: {luong}");
            Console.WriteLine("Loai hop dong: {0}", loaiHD ? "Dai han" : "Ngan han");
        }

        public override bool laNVHDDH()
        {
            return loaiHD;
        }

        public override float layLuong()
        {
            return loaiHD ? luong + 2000000 : luong;
        }

        public override string loaiNV()
        {
            return "NVHD";
        }
    }
}
